### ISSUE TITLE
[v1.8.1] Release Tracker

### ISSUE THREAD

**Issue by seemethere** [pytorch/pytorch#53572]
We cut a [release branch](https://github.com/pytorch/pytorch/tree/release/1.8) for the 1.8.1 release.

Our plan from this point from this point is roughly:

* Phase 1 (until TBD): work on finalizing the release branch
* Phase 2 (after TBD): perform extended integration/stability/performance testing based on Release Candidate builds.

This issue is for tracking cherry-picks to the release branch.

## Cherry-Pick Criteria

**Phase 1 (until TBD):**

Only low-risk changes may be cherry-picked from master:

1. Fixes to regressions against the most recent minor release (e.g. 1.8.1 for 1.8.x release; see [topic: regression issue list](http://github.com/pytorch/pytorch/issues?q=is:issue%20is:open%20label:%22topic:%20regression%22/))
2. Critical fixes for: [silent correctness](http://github.com/pytorch/pytorch/issues?q=is:issue%20is:open%20label:%22topic:%20correctness%20(silent)"/), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](http://is:issue%20is:open%20label:%22topic:%20deadlock"/), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+).
3. Fixes to new features introduced in the most recent minor release (e.g. 1.8.1 for 1.8.x release)
4. Test/CI fixes
5. Documentation improvements
6. Release branch specific changes (e.g. change version identifiers)

Any other change requires special dispensation from the release managers (currently [@malfet](https://github.com/malfet), [@seemethere](https://github.com/seemethere), [@gchanan](https://github.com/gchanan)).   If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain.

**Phase 2 (after TBD):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are:

**Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

This will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/1.8).
3. Make a request below with the following format:

```
Link to landed master PR (if applicable):
Link to release branch PR:
Criteria Category:`
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **NOTE: do not land the release branch PR yourself. Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

**Comment by gmagogsfm** [comment]
**Cherrypick request:**
Link to landed master PR (if applicable): #53576
Link to release branch PR: #53667
Criteria Category:`3. Fixes to new features introduced in the most recent minor release (e.g. 1.8.1 for 1.8.x release)

CC @jamesr66a

**Comment by malfet** [comment]
Link to landed master PR: #48570
Link to release branch PR: #53671
Criteria Category: Fixes to new features introduced in the most recent minor release

**Comment by jamesr66a** [comment]
Link to landed master PR (if applicable): #51928 79944f7 #53250 #53630
Link to release branch PR: https://github.com/pytorch/pytorch/pull/53674
Criteria Category: Documentation improvements

**Comment by malfet** [comment]
Link to landed master PR: #52339
Link to release branch PR: #53675
Criteria Category: Special dispensation

**Comment by seemethere** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/issues/53133
Link to release branch PR: https://github.com/pytorch/pytorch/pull/53734
Criteria Category: Test/CI fixes (resolves https://github.com/pytorch/pytorch/issues/52662)

**Comment by gmagogsfm** [comment]
> **Cherrypick request:**
> Link to landed master PR (if applicable): #53576
> Link to release branch PR: #53766
> Criteria Category:`3. Fixes to new features introduced in the most recent minor release (e.g. 1.8.1 for 1.8.x release)
> 
> Details: FX is introduced in 1.8, which emits `Ellipsis` constant when generating Python code, this cherrypick is needed to ensure FX-generated code is TorchScript-able as promised in release notes.
> 
> CC @jamesr66a
> 
> @malfet: Can you please clarify what new features are you talking about?

@malfet Just added more details

**Comment by neerajprad** [comment]
Link to landed master PR (if applicable): #52763 #53600
Link to release branch PR: #53763
Criteria Category:  Fixes to new features introduced in the most recent minor release [Minor fixes to distributions validation check which is enabled by default in torch 1.8.0.]

**Comment by lw** [comment]
Link to landed master PR: N/A
Link to release branch PR: #53804
Criteria category: crashes

**Comment by mrshenli** [comment]
Link to landed master PR: #53508
Link to release branch PR: #53822
Criteria category: Doc-only fixes

**Comment by malfet** [comment]
Link to the master PR: #53447 
Link to the release branch PR: #53986 
Criteria category: Fixes regression from previous release

**Comment by malfet** [comment]
Link to master PR: #53267
Link to release branch PR: #53987
Criteria category: CI sanity

**Comment by zou3519** [comment]
Link to landed master PR (if applicable): #51983
Link to release branch PR: #54003
Criteria Category: Documentation improvements

**Comment by malfet** [comment]
Link to landed master PR: #53548
Link to release branch PR: #54005
Criteria Category: ONNX fixes

**Comment by malfet** [comment]
Link to landed master PR: #53529
Link to release branch PR: #54007
Criteria Category: ONNX fixes

**Comment by VitalyFedyunin** [comment]
Link to master PR: #53991 (waiting land)
Link to release branch PR: #54015
Criteria Category: OneDNN fixes

**Comment by jbschlosser** [comment]
> Link to the master PR: #53447
> Link to the release branch PR: #53986
> Criteria category: Fixes regression from previous release

@malfet There's a related doc-only change for this regression fix: https://github.com/pytorch/pytorch/pull/53809

**Comment by BowenBao** [comment]
Link to landed master PR: #53311
Link to release branch PR: #54019
Criteria Category: ONNX fixes for #53730

cc @malfet

**Comment by malfet** [comment]
Link to landed master PR: #52616
Link to release branch PR: #54022
Criteria Category: Fixes to new features introduced in the most recent minor release (aarch64 support)

**Comment by malfet** [comment]
Link to landed master PR: #53871
Link to release branch PR: #54025
Criteria Category: Regression from the previous release (crashes in DataLoader)

**Comment by malfet** [comment]
Link to landed master PR: #53809
Link to release branch PR: #54026
Criteria Category: Doc fixes

**Comment by malfet** [comment]
Link to landed master PR: #52216
Link to release branch PR: #54028
Criteria Category: PR fixes (XLA followup for #48570)

**Comment by BowenBao** [comment]
Link to landed master PR: #53931
Link to release branch PR: #54033
Criteria Category: ONNX fixes for embedding updates

cc @malfet

**Comment by ilia-cher** [comment]
Link to release branch PR: https://github.com/pytorch/pytorch/pull/54044
(master PRs are landed in pytorch/kineto)
Criteria Category: Kineto bugfixes

**Comment by mrshenli** [comment]
Placeholder for #51820

**Comment by seemethere** [comment]
Link to release branch PR: https://github.com/pytorch/pytorch/pull/54205
Criteria Category: Fixes build regression introduced in https://github.com/pytorch/pytorch/pull/54044

**Comment by pritamdamania87** [comment]
Link to landed master PR: https://github.com/pytorch/pytorch/pull/53929
Link to release branch PR: https://github.com/pytorch/pytorch/pull/54358
Criteria Category: Fixes bug in distributed autograd with respect to CUDA stream synchronizations (silent correctness)

**Comment by brianjo** [comment]
Link to land branch PR: https://github.com/pytorch/pytorch/pull/54141
Critical Category: Updates for our doc build for the new version.

**Comment by lw** [comment]
Link to landed master PR: N/A
Link to release branch PR: https://github.com/pytorch/pytorch/pull/54491
Criteria Category: Fixes build failure on old systems (no-op for official builds)

**Comment by malfet** [comment]
Link to landed master PR: #53841
Link to release master PR: #54537
Criteria Category: Fixes regression (torch.cuda.randperm) from previous release

**Comment by pritamdamania87** [comment]
> @malfet: This sounds a bit risky to be picked into 1.8.1 (it affects codepath for all tensor operators) Is there an issue with the repro example?

@malfet Regarding https://github.com/pytorch/pytorch/pull/53929, this only impacts distributed autograd and not vanilla PyTorch autograd. In terms of repro, there isn't an issue but @mrzzd shared a repro with me over chat and the unit test used in that PR is basically the repro of the issue. The reason I put up the PR to be included in 1.8.1 was because without it, it results in incorrect results while using distributed autograd with CUDA.

**Comment by malfet** [comment]
Link to landed master PRs:  #54431 and  #54527
Link to release branch PR: #54556
Criteria Category: CI fixes

**Comment by seemethere** [comment]
`v1.8.1` has been released, going to go ahead and close this issue as well as lock it
